Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  SignupController.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/7/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+public final class SignupController: UIViewController {
+    
+    private let _viewModel: SignupViewModelType
+    private let _signupViewFactory: () -> SignupViewType
+    private let _delegate: SignupControllerDelegate
+    
+    public lazy var signupView: SignupViewType = self._signupViewFactory()
+    
+    private lazy var _notificationCenter: NSNotificationCenter = NSNotificationCenter.defaultCenter()
```

Comment:
In theory if you want to follow dependency injection all dependencies should be passed through the constructor / initializer as interfaces / protocol so they can be mocked when testing. That being said I've never found my self in a situation that I wanted to use a different instance of `NSNotificationCenter` rather than the default one. So, I would say leave it as an internal implementation detail and use a `let`.